### PR TITLE
fix: Permission created before making device a control node is not getting active

### PR DESCRIPTION
**Pull request by kounkou**
### What does the PR do

This PR mitigates permission stuck in pending state upon making device a control node. It fixes
[#14023](https://github.com/status-im/status-desktop/issues/14023) by applying events and publishing community events once the device is promoted to control node.
Related to https://github.com/status-im/status-go/pull/5070

### Affected areas

- NA

### Screenshot of functionality (including design for comparison)

Performing tests to gather evidence for resolution.

### CODE REVIEW

**Comment by anastasiyaig** [comment]
```
panic: interface conversion: interface {} is nil, not int64

goroutine 29992 [running]:
github.com/status-im/status-go/services/wallet.(*Reader).startWalletEventsWatcher.func1({{0x10cafb0fc, 0x19}, 0x0, {0x2400711b668, 0x1, 0x1}, {0x0, 0x0}, 0x662a7d1d, 0xa, ...})
	/Users/jenkins/workspace/s_macos_aarch64_package_PR-14537/vendor/status-go/services/wallet/reader.go:225 +0x19c
github.com/status-im/status-go/services/wallet/walletevent.watch({0x10d7e1d58, 0x24004819400}, 0x2400043fdd0?, 0x2402742da60)
	/Users/jenkins/workspace/s_macos_aarch64_package_PR-14537/vendor/status-go/services/wallet/walletevent/watcher.go:61 +0x1a0
github.com/status-im/status-go/services/wallet/walletevent.(*Watcher).Start.func1({0x10d7e1d58?, 0x24004819400?})
	/Users/jenkins/workspace/s_macos_aarch64_package_PR-14537/vendor/status-go/services/wallet/walletevent/watcher.go:34 +0x30
github.com/status-im/status-go/services/wallet/async.(*Group).Add.func1()
	/Users/jenkins/workspace/s_macos_aarch64_package_PR-14537/vendor/status-go/services/wallet/async/async.go:107 +0x34
created by github.com/status-im/status-go/services/wallet/async.(*Group).Add
	/Users/jenkins/workspace/s_macos_aarch64_package_PR-14537/vendor/status-go/services/wallet/async/async.go:106 +0x88
SIGABRT: Abnormal termination.
zsh: abort      ./nim_status_client --datadir=~/data/pr1 --LOG_LEVEL=DEBUG
````

@kounkou i am seeing this in this PR which is fixed in master certainly.  Also there is 1 test failing and thats also because of some go changes being done recently (the chain ID value being sent)

can you please rebase the PR so it has recent commits from go?

**Comment by anastasiyaig** [comment]
also, i tested the scenario from the bug and it is reproduced as described
permission is not getting active even after restart

Screenshot 2024-04-29 at 17 59 54

https://github.com/status-im/status-desktop/assets/82375995/52bb7a83-3e8e-4b57-9cf8-014a350393d7

Screenshot 2024-04-29 at 17 59 45

**Comment by kounkou** [comment]
Thanks a lot for the testing, I am having another look now.

**Comment by kounkou** [comment]
Hi @anastasiyaig can I get a retest on this ticket please ?

**Comment by anastasiyaig** [comment]
sure, @glitchminer or @lukaszso could you assist please?

File: ui/app/AppLayouts/Communities/panels/OverviewSettingsFooter.qml [hunk 1]
```diff
@@ -104,6 +105,8 @@ Control {
                 size: StatusBaseButton.Size.Small
                 text: d.primaryButtonText
                 onClicked: d.primaryButtonAction()
+                interactive: root.isPromoteSelfToControlNodeEnabled
+                tooltip.text: root.isPromoteSelfToControlNodeEnabled ? "" : qsTr("Owner token is needed")
```
[the remark this answers is not in the document]
Comment (kounkou) [reply ui/app/AppLayouts/Communities/panels/OverviewSettingsFooter.qml]: > as an option - check if we show promote for admin role and do the same

I like the idea to be consistent with the `admin` UI. Updated. I still feel we could add something to let the user know what they need to do

**Comment by kounkou** [comment]
Updated, thanks

**Comment by anastasiyaig** [comment]
@kounkou you requested a test from qa here, was it done or you just merged?

**Comment by kounkou** [comment]
> @kounkou you requested a test from qa here, was it done or you just merged?

Hi anastasiyaig@ 
It was done by caybro@ however please note that the original test as stated in the ticket is no longer valid as we have added a requirement that in order to promote self as control node, we need to have minted the owner token beforehand.

**Pull request merged**